packages/mui-material/src/AvatarGroup/AvatarGroup.test.js

```diff
@@ -49,6 +49,63 @@ describe('<AvatarGroup />', () => {
     expect(container.textContent).to.equal('+2');
   });
 
+  it('should respect total', () => {
+    const { container } = render(
+      <AvatarGroup total={10}>
+        <Avatar src="/fake.png" />
+        <Avatar src="/fake.png" />
+        <Avatar src="/fake.png" />
+      </AvatarGroup>,
+    );
+    expect(container.querySelectorAll('.MuiAvatar-root').length).to.equal(4);
+    expect(container.querySelectorAll('img').length).to.equal(3);
+    expect(container.textContent).to.equal('+7');
+  });
+
+  it('should respect both total and max', () => {
+    const { container } = render(
+      <AvatarGroup max={2} total={3}>
+        <Avatar src="/fake.png" />
+        <Avatar src="/fake.png" />
+      </AvatarGroup>,
+    );
+    expect(container.querySelectorAll('.MuiAvatar-root').length).to.equal(2);
+    expect(container.querySelectorAll('img').length).to.equal(1);
+    expect(container.textContent).to.equal('+2');
+  });
+
+  it('should respect total and clamp down shown avatars', () => {
+    const { container } = render(
+      <AvatarGroup total={1}>
+        <Avatar src="/fake.png" />
+        <Avatar src="/fake.png" />
+      </AvatarGroup>,
+    );
+    expect(container.querySelectorAll('.MuiAvatar-root').length).to.equal(1);
+    expect(container.querySelectorAll('img').length).to.equal(1);
+    expect(container.textContent).to.equal('');
+  });
+
+  it('should display no avatars when total is zero', () => {
+    const { container } = render(
+      <AvatarGroup total={0}>
+        <Avatar src="/fake.png" />
+      </AvatarGroup>,
+    );
+    expect(container.querySelectorAll('.MuiAvatar-root').length).to.equal(0);
+    expect(container.textContent).to.equal('');
+  });
+
+  it('should ignore negative total', () => {
+    const { container } = render(
+      <AvatarGroup total={-1}>
+        <Avatar src="/fake.png" />
+      </AvatarGroup>,
+    );
+    expect(container.querySelectorAll('.MuiAvatar-root').length).to.equal(1);
+    expect(container.textContent).to.equal('');
+  });
```

Comment:
For me, I would neglect this case and let the developers handle it.

Reply:
removed this test case